From 8d40fbabead00847199af114fdc91f3f9a1ab397 Mon Sep 17 00:00:00 2001
From: Stuart Caie <kyzer@cabextract.org.uk>
Date: Tue, 6 Feb 2018 23:17:30 +0000
Subject: [PATCH 3/3] Fix off-by-one error in chmd TOLOWER() fallback

(cherry picked from commit 4fd9ccaa54e1aebde1e4b95fb0163b699fd7bcc8)
---
 libmspack/trunk/mspack/chmd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libmspack/trunk/mspack/chmd.c b/libmspack/trunk/mspack/chmd.c
index b799154..cea9fc2 100644
--- a/libmspack/trunk/mspack/chmd.c
+++ b/libmspack/trunk/mspack/chmd.c
@@ -834,7 +834,7 @@ static int search_chunk(struct mschmd_header *chm,
 # endif
 # define TOLOWER(x) tolower(x)
 #else
-# define TOLOWER(x) (((x)<0||(x)>256)?(x):mspack_tolower_map[(x)])
+# define TOLOWER(x) (((x)<0||(x)>255)?(x):mspack_tolower_map[(x)])
 /* Map of char -> lowercase char for the first 256 chars. Generated with:
  * LC_CTYPE=en_GB.utf-8 perl -Mlocale -le 'print map{ord(lc chr).","} 0..255'
  */
-- 
2.18.0

